### PR TITLE
JWW User Stories 21 & 22

### PR DESCRIPTION
**Pull request by iEv0lv3**
##### Edit: Completely changed the way updating passwords and user data is handled. Previously all updates were done through the `users_controller` `update` action and required a cascade of if/elsif statements in order to test all possible edge cases. Updating passwords is now under its own controller within `profile` called `security_controller`, which handles all functionality related to passwords. The `profile` `users_controller` `update` action now only handles updating user profile data and not passwords.

Adds functionality to edit user profile information without updating the password, and additional functionality to update a user password independently from the other user profile information.

Completes User Story #
20, 21, 22

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Notes

Ray's updates are included in this PR.

## RSpec Results
```
Finished in 8.13 seconds (files took 2.63 seconds to load)
145 examples, 0 failures

Coverage report generated for RSpec to /Users/jww/turing/2module/projects/monster_shop_1911/coverage. 2126 / 2126 LOC (100.0%) covered.
```

### CODE REVIEW

File: app/controllers/profile/users_controller.rb [hunk 1]
```diff
@@ -9,15 +9,26 @@ def edit
...
-    if @user.update(user_params)
+    current_pw = @user.password
+    if @user.update(user_params) && current_pw == @user.password
       flash[:success] = 'Profile successfully updated'
       redirect_to '/profile'
+    elsif @user.update(user_params) && current_pw != @user.password
+      flash[:success] = 'Password updated'
+      redirect_to '/profile'
+    elsif user_params[:password] != user_params[:password_confirmation]
+      flash[:error] = @user.errors.full_messages.to_sentence
```
Comment (raymondknguyen) [inline app/controllers/profile/users_controller.rb]: may have a helper method for this... park it for revisions

**Review by raymondknguyen — APPROVED** [review]

**Review by d-atkins — COMMENTED** [review]
bug: leaving both fields blank when editing password doesn't change the password but displays a "Profile successfully updated" flash message

File: app/controllers/profile/users_controller.rb (hunk 1 above)
Comment (iEv0lv3) [reply app/controllers/profile/users_controller.rb]: Agree, and I completely re-coded the entire logic just now. I used encapsulation to handle the password updates and user profile data updates in different controllers.

**Review by d-atkins — APPROVED** [review]
